extractor/src/main/java/org/schabi/newpipe/extractor/stream/Stream.java

```diff
@@ -1,108 +1,208 @@
 
     /**
-     * Check if the list already contains one stream with equals stats
+     * Gets the ID for this stream, e.g. itag for YouTube.
+     *
+     * @return the id
      */
-    public static boolean containSimilarStream(Stream stream, List<? extends Stream> streamList) {
-        if (isNullOrEmpty(streamList)) return false;
-        for (Stream cmpStream : streamList) {
-            if (stream.equalStats(cmpStream)) return true;
-        }
-        return false;
+    public String getId() {
+        return id;
     }
 
     /**
-     * Gets the url.
+     * Gets the URL.
      *
-     * @return the url
+     * @return the URL if the content is a URL, {@code null} otherwise
+     * @deprecated Use {@link #getContent()} instead
      */
+    @Deprecated
+    @Nullable
     public String getUrl() {
-        return url;
+        return isUrl ? content : null;
     }
 
     /**
-     * Gets the torrent url.
+     * Gets the content or URL.
      *
-     * @return the torrent url, example https://webtorrent.io/torrents/big-buck-bunny.torrent
+     * @return the content or URL
      */
-    public String getTorrentUrl() {
-        return torrentUrl;
+    public String getContent() {
+        return content;
     }
 
     /**
-     * Gets the format.
+     * Returns if the content is a URL or not.
+     *
+     * @return {@code true} if the content of this stream content is a URL, {@code false}
+     * if it is the actual content
+     */
+    public boolean isUrl() {
+        return isUrl;
+    }
+
+    /**
+     * Gets the {@link MediaFormat}, which can be null.
      *
      * @return the format
      */
+    @Nullable
     public MediaFormat getFormat() {
         return mediaFormat;
     }
 
     /**
-     * Gets the format id.
+     * Gets the format id, which can be unknown.
      *
-     * @return the format id
+     * @return the format id or {@link #FORMAT_ID_UNKNOWN}
      */
     public int getFormatId() {
-        return mediaFormat.id;
+        if (mediaFormat != null) {
+            return mediaFormat.id;
+        }
+        return FORMAT_ID_UNKNOWN;
+    }
+
+    /**
+     * Gets the delivery method.
+     *
+     * @return the delivery method
+     */
+    @Nonnull
+    public DeliveryMethod getDeliveryMethod() {
+        return deliveryMethod;
     }
+
+    /**
+     * Gets the base URL of a stream.
+     * <p>
+     * If the stream is not a DASH stream or an HLS stream, this value will always be null.
+     * It may be also null for these streams too.
+     * </p>
+     *
+     * @return the base URL of the stream or {@code null}
+     */
+    @Nullable
+    public String getBaseUrl() {
+        return baseUrl;
+    }
+
+    /**
+     * Gets the {@link ItagItem} of a stream.
+     * <p>
+     * If the stream is not a YouTube stream, this value will always be null.
+     * </p>
+     *
+     * @return the {@link ItagItem} of the stream or {@code null}
+     */
+    @Nullable
+    public abstract ItagItem getItagItem();
 }
```

Comment:
Referencing https://github.com/TeamNewPipe/NewPipe/pull/6537#discussion_r780705875: just ask Android Studio to generate the `hashCode()` method here and in `VideoStream` and `AudioStream` (it will probably also autogenerate `equals()` since it's good practice to implement either none or both methods)

Reply:
Done, I also generated them for `SubtitlesStream`. Thank you!

Reply:
(I don't think you have pushed the PR; ignore this if that's expected)

Reply:
Done